src/trees/prufer.jl

```diff
@@ -0,0 +1,91 @@
+"""
+    is_tree(g)
+
+Returns true if g is a tree: that is, a simple, connected undirected graph, with nv-1 edges (nv = number of vertices). Trees are the minimal connected graphs; equivalently they have no cycles.  
+
+This function does not apply to directed graphs. Directed trees are sometimes called [polytrees](https://en.wikipedia.org/wiki/Polytree)). 
+
+"""
+
+function is_tree(g::AbstractGraph)
+    is_directed(g) && throw(ArgumentError("g must be undirected. "))
```

Comment:
We currently use the `@traitfn` macro to restrict functions to directed/undirected graphs. An example can be found here: https://github.com/JuliaGraphs/Graphs.jl/blob/ea6bcfeb71de9d1229bbd411ef9b10cac44fa795/src/traversals/dfs.jl#L14